app/models.py

```diff
@@ -1213,6 +1215,15 @@ def has_used_alias_from_partner(self) -> bool:
             > 0
         )
 
+    def is_domain_blocked(self, domain: str) -> bool:
+        """checks whether the provided domain is blocked for a given user"""
+        domain_names = []
+
+        for blocked_domain in BlockedDomain.filter_by(user_id=self.id):
+            domain_names.append(blocked_domain.domain)
```

Comment:
Can you add the domain to the filter directly? This you skip a loop and the filtering happens DB side directly.